Address single-threadedness in FAQ (#4379)
dart-lang/site-www: src/_guides/language/language-tour.md

Instead of threads, all Dart code runs inside of *isolates*. 

> Each Dart isolate has a single thread of execution and
> shares no mutable objects with other isolates.
> Spinning up multiple isolates creates multiple threads of execution,
> enabling multi-threading without the typical issues that surround it.

Reviewer: **suggestion:** 

```suggestion
Each Dart isolate uses one execution thread and
shares no mutable objects with other isolates.
Spinning up multiple isolates creates multiple threads of execution.
This enables multi-threading without the typical issues that surround it.
```

**question:** What issues?

Reviewer 2: The primary issue is race conditions, e.g.: https://en.wikipedia.org/wiki/Race_condition#In_software

Author: > Each Dart isolate uses one execution thread and

Going to keep "a single thread of execution" as is to match the language of the original issue.

concurrency is error prone and can lead to complicated code.

Instead of threads, all Dart code runs inside of *isolates*. 
Each Dart isolate uses a single thread of execution and
shares no mutable objects with other isolates.
Spinning up multiple isolates creates multiple threads of execution.

> This enables multi-threading without its primary drawback, race conditions.

Reviewer 2: Should we link `race conditions` to the wikipedia page? I find their example pretty easy to read.
